pkg/cli/idle/idle.go

```diff
@@ -593,17 +597,44 @@ func (o *IdleOptions) RunIdle() error {
 		if err != nil {
 			return err
 		}
-		b.ResourceNames("endpoints", targetServiceNames...)
+
+		var endpointSliceNames []string
+		for _, val := range targetServiceNames {
+			endpointSlices, err := o.ClientSet.DiscoveryV1().EndpointSlices(o.Namespace).List(context.Background(), metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("%s=%s", v1.LabelServiceName, val),
+			})
+			if err != nil {
+				return err
+			}
+			for _, slice := range endpointSlices.Items {
+				endpointSliceNames = append(endpointSliceNames, slice.Name)
+			}
+		}
+
+		b.ResourceNames("endpointslices", endpointSliceNames...)
```

Comment:
(I'm not convinced that this whole "visitor pattern" thing is really a simplification rather than a complication at this point..)